### ISSUE TITLE
Priority not being respected

### ISSUE THREAD

**Issue by ghost** [nzbget/nzbget#305]
If I have a download with normal priority downloading in nzbget.. Say its going to take 20 hours to complete said download.

I have another app that drops off nzb's to nzbget with priority very_high.. 

these get added to the nzbget queue with the appropriate priority but this priority doesnt affect anything until the next queue item is selected for download which is going to be 20 hours, when the existing normal priority download finishes.

Shouldnt nzbget pause the existing normal priority download and start the very high priority download immediately when it is added... 

I can see if both were the same priority -- .. but since the added download is a higher priority then the one currently downloading, the exiting download should be paused IMO.

Please let me know if you require further information.

Thanks

**Comment by hugbug** [comment]
The higher priority items should be downloading first. The lower priority items isn't paused however (that's not necessary).

Please provide info about your nzbget version, OS and where you have obtained nzbget.
Please also post a screenshot (you may want to edit titles first).

**Comment by ghost** [comment]
Do categories effect priority? Note - in the attached screenshot. The "normal" priority item is tagged with "movies" category, while the higher priority paused item is tagged with "series" priority.
http://imgur.com/a/sdqMz

I am using: 18.0-testing-r1820 as reported by the "System" tab in settings within nzbget itself. I believe I obtained nzbget directly from the nzbget website but it was quite some time ago. I am running on windows.

Please let me know if you require more information.

Thanks

**Comment by ghost** [comment]
ok so i did a little more investigation.. the issue has to do with the "completion.py" script that is being run on scan and add to queue

this is a description of the script: Verifies that enough articles are available before starting the download.
This script should be added as:
- Scan script: to pause incoming NZBs.
- Queue script: to check a newly added, and by the script paused NZB, resume when OK.
- Scheduler script: to regularly check completeness of by the script paused NZB in the queue, will resume when OK or mark as BAD after file is X hours old.

NOTE: To stop the script when checking, reload NZBget.

Info about the NZBGet Completion.py script:
Author: kloaknet.
Support and updates: http://nzbget.net/forum/viewtopic.php?f=8&t=1736.
Date: Oct 22th, 2016.
License: GPLv3 (http://www.gnu.org/licenses/gpl.html).
PP-script Version: 0.2.8.

NOTE: This script requires NZBget 13.0+ and Python 2.7.9+ to be installed on your system.

**Comment by ghost** [comment]
the page for the script is here": http://forum.nzbget.net/viewtopic.php?f=8&t=1736

I'll post a reply there and see what they say.

Thanks!

**Comment by hugbug** [comment]
On your screenshot the second item is in paused state., That's why it is not downloaded.

The completion script indeed controls which items are downloaded or not by pausing/unpausing them. Try to look into log if you see any messages from the completion script. In a case you can't resolve the issue please contact the author of the script via forum topic.
